sql/core/src/test/scala/org/apache/spark/sql/JoinSuite.scala

```diff
@@ -141,6 +141,7 @@ class JoinSuite extends QueryTest with SharedSQLContext {
       ("SELECT * FROM testData right join testData2 ON key = a and key = 2",
         classOf[BroadcastHashJoinExec])
     ).foreach(assertJoin)
+    sql("UNCACHE TABLE testData2")
     sql("UNCACHE TABLE testData")
```

Comment:
Instead doing this, we should use the finally block with `spark.sharedState.cacheManager.clearCache()`